File: app/src/main/java/org/digitalcampus/oppia/utils/custom_prefs/AdminSimplePreference.java
package org.digitalcampus.oppia.utils.custom_prefs;

import android.content.Context;
import android.util.AttributeSet;

import androidx.preference.EditTextPreference;
import androidx.preference.Preference;

public class AdminSimplePreference extends Preference implements AdminPreference {

    private OnPreferenceClickListener customOnPreferenceClickListener;

    public AdminSimplePreference(Context context) {
        super(context);
    }

    public AdminSimplePreference(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public AdminSimplePreference(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }


    @Override
    public void setOnPreferenceClickListener(OnPreferenceClickListener onPreferenceClickListener) {
        super.setOnPreferenceClickListener(onPreferenceClickListener);
    }

    @Override
    public void onAccessGranted() {
        super.setOnPreferenceClickListener(null);
        super.performClick();
    }
}

File: app/src/main/java/org/digitalcampus/oppia/utils/custom_prefs/AdminSimplePreference.kt
package org.digitalcampus.oppia.utils.custom_prefs

import android.content.Context
import android.util.AttributeSet
import androidx.preference.Preference

class AdminSimplePreference : Preference, AdminPreference {
    private val customOnPreferenceClickListener: OnPreferenceClickListener? = null

    constructor(context: Context?) : super(context!!) {}
    constructor(context: Context?, attrs: AttributeSet?) : super(context!!, attrs) {}
    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(context!!, attrs, defStyleAttr) {}

    override fun setOnPreferenceClickListener(onPreferenceClickListener: OnPreferenceClickListener?) {
        super.setOnPreferenceClickListener(onPreferenceClickListener)
    }

    override fun onAccessGranted() {
        super.setOnPreferenceClickListener(null)
        super.performClick()
    }
}